atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt

```diff
@@ -1242,9 +1245,14 @@ class AtomicFUTransformer(
 
         // generates a ref class with volatile field of primitive type inside
         private fun generateRefVolatileClass(f: FieldInfo, arg: Type) {
+            val visibility = when {
+                f.hasExternalAccess -> ACC_PUBLIC
+                f.accessors.isEmpty() -> ACC_PRIVATE
```

Comment:
This is the visibility of a _class_ and class cannot be `PRIVATE`. Please, add a test for this case. You can get rid of accessors by using `@JvmField` annotation on a top-level val. Class visibility will need its own code (separate from field visibility)

Also, this code should be placed after `if (analyzePhase2) return // nop` line below.

Reply:
Fixed this added `*RefVolatile` class modifiers check for all static field reflection tests.